Code/src/main/java/nl/utwente/group10/ui/components/blocks/Block.java

```diff
@@ -2,29 +2,35 @@
 
 import java.io.IOException;
 
+import javafx.scene.input.MouseButton;
 import javafx.scene.input.MouseEvent;
 import javafx.scene.layout.StackPane;
+
 import nl.utwente.group10.haskell.expr.Expr;
 import nl.utwente.group10.ui.CustomUIPane;
 import nl.utwente.group10.ui.components.ComponentLoader;
 import nl.utwente.group10.ui.components.anchors.OutputAnchor;
+import nl.utwente.group10.ui.menu.CircleMenu;
 
 /**
- * Base UI Component that other visual elements will extend from. If common functionality is found it should be
- * refactored to here.
+ * Base UI Component that other visual elements will extend from. If common
+ * functionality is found it should be refactored to here.
  */
 public abstract class Block extends StackPane implements ComponentLoader {
 
-    /** The output of this Block.**/
+    /** The output of this Block. **/
     private OutputAnchor output;
-
     /** The pane that is used to hold state and place all components on. */
     private CustomUIPane parentPane;
+    /** The context menu associated with this block instance. */
+    private CircleMenu contextMenu;
```

Comment:
I'd call it a circle menu consistently to, again, avoid confusion with the main menu which is a ContextMenu.